ride/lp.ride

```diff
@@ -795,6 +797,66 @@ func mustManager(i: Invocation) = {
   }
 }
 
+@Callable(i)
+func calculateAmountOutForSwapREADONLY(cleanAmountIn: Int,  isReverse: Boolean) = {
+let (assetOut, assetIn) = if (isReverse == false) then {
+    let assetOut = getStringOrFail(this, pa())
+    let assetIn = getStringOrFail(this, aa())
+    (assetOut, assetIn)
+  } else {
+    let assetOut = getStringOrFail(this, aa())
+    let assetIn = getStringOrFail(this, pa())
+    (assetOut, assetIn)
+  }
+  let poolAssetInBalance = getAccBalance(assetIn)
+  let poolAssetOutBalance = getAccBalance(assetOut)
+  let amountOut = fraction(poolAssetOutBalance, cleanAmountIn, poolAssetInBalance + cleanAmountIn)
+
+  let oldK = poolAssetInBalance.toBigInt() * poolAssetOutBalance.toBigInt()
+  let newK = (getAccBalance(assetIn).toBigInt() + cleanAmountIn.toBigInt()) * (getAccBalance(assetOut).toBigInt() - amountOut.toBigInt())
+  strict checkK = newK >= oldK || "new K is fewer error".throw()
+
+  (nil, amountOut)
+}
+
+@Callable(i)
+func calculateAmountOutForSwapAndSendTokens(cleanAmountIn: Int, isReverse: Boolean, amountOutMin: Int, addressTo: String) = {
+  let swapContact = invoke(
+    factoryContract,
+    "getSwapContractREADONLY", 
+    [], 
+    []
+  ).exactAs[String]
+  strict checks = [
+    i.payments[0].value().amount >= cleanAmountIn || "Wrong amount".throwErr(),
```

Comment:
Looks strange. `cleanAmountIn` should be equal to `i.payments[0].value().amount` so the argument `cleanAmountIn` does not needed at all.

Reply:
I use >= in case if we going to change attached payment with fee included

Reply:
You just approved there is no reason to have it now. And maybe there will be no reason in the long run. But it here forever. Ok.